minidump/src/context.rs

```diff
@@ -80,15 +88,80 @@ pub trait CpuContext {
         )
     }
 
+    /// An iterator over all registers in this context.
+    ///
+    /// This iterator yields registers and values regardless of whether the register is valid. To
+    /// get valid values, use [`valid_registers`](Self::valid_registers), instead.
+    fn registers(&self) -> CpuRegisters<'_, Self> {
+        self.valid_registers(&MinidumpContextValidity::All)
+    }
+
+    /// An iterator over valid registers in this context.
+    ///
+    /// This iterator yields valid registers and their values.
+    fn valid_registers<'a>(&'a self, valid: &'a MinidumpContextValidity) -> CpuRegisters<'a, Self> {
+        let regs = match valid {
+            MinidumpContextValidity::All => CpuRegistersInner::Slice(Self::REGISTERS.iter()),
+            MinidumpContextValidity::Some(valid) => CpuRegistersInner::Set(valid.iter()),
+        };
+
+        CpuRegisters {
+            regs,
+            context: self,
+        }
+    }
+
     /// Gets the name of the stack pointer register (for use with get_register/set_register).
     fn stack_pointer_register_name(&self) -> &'static str;
+
     /// Gets the name of the instruction pointer register (for use with get_register/set_register).
     fn instruction_pointer_register_name(&self) -> &'static str;
 }
 
+/// Default implementation for `CpuContext::memoize_register`.
+fn default_memoize_register(registers: &[&'static str], reg: &str) -> Option<&'static str> {
+    let idx = registers.iter().position(|val| *val == reg)?;
+    Some(registers[idx])
```

Comment:
```suggestion
    registers.iter().find(|val| val.eq_ignore_ascii_case(reg))
```
Maybe that needs some more deref magic though

Reply:
You can't use `.find()` because that would not return `'static`

Reply:
We haven't actually found a reason to need this, so let's not apply this suggestion.